### PR TITLE
Add Upgrade Notice re Cassandra num_tokens setting

### PR DESCRIPTION
**Pull request by johnsmartco**
**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CHANGELOG.md updated (not required for documentation PRs)
- [ ] CLA Signed:  [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: docs/content/en/release-notes/_index.md [hunk 1]
```diff
@@ -57,9 +57,92 @@ The K8ssandra helm chart deploys the following components. Some are optional, an
...
+    enabled: true
+  heap:
+    size: 512M
+  datacenters:
+  - name: dc1
+    size: 1
+    num_tokens: 128
+```
+
+(We're using `num_tokens: 128` above just as an example.)
```
Comment (jdonenine) [inline docs/content/en/release-notes/_index.md:101]: I think that if they did not previously set it, they will have 256, the default, so we should instruct them to manually set it to 256 so that it is forced to match.

**Review by jdonenine — CHANGES_REQUESTED** [review]
Thanks for picking this up so quickly @johnsmartco .

We will hopefully have a 1.3.1 patch out quickly that will address the problem as well.

**Comment by jdonenine** [comment]
@johnsmartco @jsanda I did a quick test to confirm that the work around does what we want (@jsanda talked about testing this out earlier) and things do work as long you set the `num_tokens` as described.  

We should definitely instruct folks to set it to 256 in the cases where they did not previous specify it directly, because that will be the default coming from 3.11.

Here's what I tested out:

## Initial installation

Installs k8ssandra-1.3.0 w/ cassandra-3.11.10 and the default num_tokens (256)

(Note that startgate is disabled here just for the sake of speed of starting the full deployment)

```
% helm install k8ssandra -f k8ssandra.values.yml k8ssandra-stable/k8ssandra
```

k8ssandra.values.yml
```
cassandra:
  version: "3.11.10"
  datacenters:
    - 
      name: dc1
      size: 1
stargate:
  enabled: false
```

### Resulting deployment

```
% helm ls
NAME            NAMESPACE       REVISION        UPDATED                                 STATUS          CHART           APP VERSION
k8ssandra       default         1               2021-08-17 18:58:03.831212 -0400 EDT    deployed        k8ssandra-1.3.0 
```

```
% kd cassdc dc1 | grep "Cassandra Operator Progress:"                              
  Cassandra Operator Progress:  Ready
```

```
% kd cassdc dc1 | grep "Server Version:"             
  Server Version:  3.11.10
```

```
% kd cassdc dc1 | grep "num_tokens:"
      num_tokens:                         256
```

## Upgrade

Upgrades to cassandra-4.0.0 and sets the num_tokens to match the previous default (256)

```
% helm upgrade k8ssandra -f k8ssandra-upgrade.values.yml k8ssandra-stable/k8ssandra
```

k8ssandra-upgrade.values.yml
```
cassandra:
  version: "4.0.0"
  datacenters:
    - 
      name: dc1
      size: 1
      num_tokens: 256
stargate:
  enabled: false
```

### Resulting Deployment

```
% kd cassdc dc1 | grep "Cassandra Operator Progress:"                              
  Cassandra Operator Progress:  Ready
```

```
% kg pods
NAME                                                READY   STATUS      RESTARTS   AGE
k8ssandra-cass-operator-5bbf6584b9-pgldj            1/1     Running     0          9m23s
k8ssandra-crd-upgrader-job-k8ssandra-skrjh          0/1     Completed   0          2m6s
k8ssandra-dc1-default-sts-0                         2/2     Running     0          106s
k8ssandra-grafana-679b4bbd74-ngrz2                  2/2     Running     0          9m23s
k8ssandra-kube-prometheus-operator-85695ffb-qhqc7   1/1     Running     0          9m23s
k8ssandra-reaper-694cf49b96-jgnvg                   0/1     Running     1          6m12s
k8ssandra-reaper-operator-5c6cc55869-tdhjw          1/1     Running     0          9m23s
prometheus-k8ssandra-kube-prometheus-prometheus-0   2/2     Running     1          9m14s
```

```
% kd cassdc dc1 | grep "Server Version:"                                           
  Server Version:  4.0.0
```

```
% kd cassdc dc1 | grep "num_tokens:"    
      num_tokens:                                    256
```

File: docs/content/en/release-notes/_index.md (hunk 1 above)
Comment (johnsmartco) [reply docs/content/en/release-notes/_index.md:101]: Good point. Updating.

**Comment by johnsmartco** [comment]
In case the edited Diffs are hard to follow, also see the latest viewed file:

https://github.com/k8ssandra/k8ssandra/blob/3fe2efac371f58ff7d70125173b689ab3ff699bc/docs/content/en/release-notes/_index.md

**Comment by johnsmartco** [comment]
Side note: the Support/Field guys requested that we separate sample command output from the actual command, for easier copy/paste of just the command. Thus the somewhat cumbersome markup with **Output**: in the doc.

File: docs/content/en/release-notes/_index.md [hunk 2]
```diff
@@ -57,9 +57,168 @@ The K8ssandra helm chart deploys the following components. Some are optional, an
...
+k8ssandra.values.yml:
+
+```yaml
+cassandra:
+  version: "3.11.10"
+  datacenters:
+    - 
+      name: dc1
+      size: 1
+stargate:
```
Comment (jdonenine) [inline docs/content/en/release-notes/_index.md:97]: You can probably just take the stargate part of the example here, then you don't need to include the note like I mentioned in my comments earlier.  It'll remove a little bit of confusion probably.

File: docs/content/en/release-notes/_index.md [hunk 3]
```diff
@@ -57,9 +57,168 @@ The K8ssandra helm chart deploys the following components. Some are optional, an
...
+
+```yaml
+cassandra:
+  version: "4.0.0"
+  datacenters:
+    - 
+      name: dc1
+      size: 1
+      num_tokens: 256
+stargate:
```
Comment (jdonenine) [inline docs/content/en/release-notes/_index.md:162]: Same comment here, we can simplify a bit just by taking this out of the example.

File: docs/content/en/release-notes/_index.md [hunk 4]
```diff
@@ -57,9 +57,168 @@ The K8ssandra helm chart deploys the following components. Some are optional, an
...
+
+```bash
+kubectl describe cassdc dc1 | grep "num_tokens:"
+```
+**Output**:
+```
+      num_tokens:                         256
+```
+
+Because the explicitly set `num_tokens` value in 4.0 matches the previous `num_tokens` default in 3.11 (256), Cassandra Operator starts.
```
Comment (jdonenine) [inline docs/content/en/release-notes/_index.md:218]: It may be more correct here to say that `Cassandra` starts, rather than the `Operator`

File: docs/content/en/release-notes/_index.md [hunk 5]
```diff
@@ -57,9 +57,168 @@ The K8ssandra helm chart deploys the following components. Some are optional, an
...
 
 {{% alert title="Important!" color="warning" %}}
+Before upgrading to K8ssandra 1.3.0, be sure to read the sections below.
+{{% /alert %}}
+
+### Upgrading to K8ssandra 1.3.0 and Cassandra 4.0
+
+When you upgrade from a prior K8ssandra release to 1.3.0, the default `cassandra.version` is now `4.0.0`. A changed `num_tokens` default may prevent Cassandra 4.0 from starting. This section explains how to avoid the issue. 
+
+First, some background information. The default value for `num_tokens` in Cassandra 3.11 is 256. The default in Cassandra 4.0 is 16. The `num_tokens` setting defines the number of tokens randomly assigned to this node on the ring. The more tokens, relative to other nodes, the larger the proportion of data that this node will store.
```
Comment (jsanda) [inline docs/content/en/release-notes/_index.md:70]: I don't know about this statement:

>The more tokens, relative to other nodes, the larger the proportion of data that this node will store.

All nodes are assigned the same number of tokens. More tokens means that the ring is divided up into smaller slices or ranges but the overall size of the ring does not change. It is true though that a larger number of tokens means that each individual range will be smaller. I don't know that we need to get into that here though.

Comment (johnsmartco) [reply docs/content/en/release-notes/_index.md:70]: I'll edit the introductory background info and keep it brief.

File: docs/content/en/release-notes/_index.md (hunk 2 above)
Comment (johnsmartco) [reply docs/content/en/release-notes/_index.md:97]: Will do

File: docs/content/en/release-notes/_index.md (hunk 3 above)
Comment (johnsmartco) [reply docs/content/en/release-notes/_index.md:162]: ok

File: docs/content/en/release-notes/_index.md (hunk 4 above)
Comment (johnsmartco) [reply docs/content/en/release-notes/_index.md:218]: Good - thx!

**Review by jsanda — APPROVED** [review]
